Issue #1515 fix xorigin yorigin disv

JoerivanEngelen opened the pull request:
Fixes #1515

# Description
Adds the following:

- Hardcode xorigins and yorigins in DISV package to 0.0.
- Add tests to verify if the xorigins and yorigins and DIS and DISV packages are rendered correctly
- Offset the grid in the DIS and DISV unittests, to verify if xorigins and yorigins are properly rendered.
- Account for xorigin and yorigin in the disv grb read funtion
- Add a test which has an offset in DISV, to see if our read DISV grb function actually properly deals with that. This is in the case somebody wants to read model output of a DISV model generated with Flopy that has an offset in xorigins and yorigins.
- Small refactor to override ``_get_render_dictionary`` instead of ``render`` in DIS and DISV
- Small refactor, move append of vertices and cell2d data to blockfile to a separate method.

# Checklist

- [x] Links to correct issue
- [x] Update changelog, if changes affect users
- [x] PR title starts with ``Issue #nr``, e.g. ``Issue #737``
- [x] Unit tests were added
- [ ] **If feature added**: Added/extended example

Huite commented:
I would strongly suggest to also fix this:

https://github.com/Deltares/imod-python/blob/afcf5412b7941ce62dc02d64e8c42c86752c570e/imod/mf6/out/disv.py#L56
https://github.com/Deltares/imod-python/blob/afcf5412b7941ce62dc02d64e8c42c86752c570e/imod/mf6/out/disv.py#L57

Then add the values to the vertices:

https://github.com/Deltares/imod-python/blob/afcf5412b7941ce62dc02d64e8c42c86752c570e/imod/mf6/out/disv.py#L63


Note that we're skipping the centroid coordinates as well:

https://github.com/Deltares/imod-python/blob/afcf5412b7941ce62dc02d64e8c42c86752c570e/imod/mf6/out/disv.py#L64
https://github.com/Deltares/imod-python/blob/afcf5412b7941ce62dc02d64e8c42c86752c570e/imod/mf6/out/disv.py#L65

This should be the least controversial since a centroid is a centroid (for a convex polygon), but as far as I remember, xugrid won't respect these coordinates anyway:
https://github.com/Deltares/xugrid/issues/56

Manangka commented on imod/mf6/disv.py:
```diff
@@ -104,10 +104,10 @@ def __init__(self, top, bottom, idomain, validate: bool = True):
-        d["xorigin"] = grid.node_x.min()
-        d["yorigin"] = grid.node_y.min()
+        d["xorigin"] = 0.0
+        d["yorigin"] = 0.0
```
I'm a bit confused. When partitioning a model, the partioned models should have same the same xorigin and yorigin as the original model right? I don't see where that is being done

JoerivanEngelen replied:
The DISV package requires explicit coordinates of where the vertices are located in the VERTICES block in its MODFLOW6 input file. We supply the UgridDataArray vertices here, which contain the full coordinates, so xorigin and yorigin need to be 0.0, as the DISV package contains a direct representation of our grid coordinates.

The DIS package, however, doesn't have such a block, only "DELR" and "DELC" data in the GRIDDATA block. We therefore rely on xorigin an yorigin here to do the bookkeeping of the coordinates.

JoerivanEngelen commented:
> I would strongly suggest to also fix this while you're at it:
> 
> https://github.com/Deltares/imod-python/blob/afcf5412b7941ce62dc02d64e8c42c86752c570e/imod/mf6/out/disv.py#L56
> 
> 
> https://github.com/Deltares/imod-python/blob/afcf5412b7941ce62dc02d64e8c42c86752c570e/imod/mf6/out/disv.py#L57
> 
> Then add the values to the vertices:
> 
> https://github.com/Deltares/imod-python/blob/afcf5412b7941ce62dc02d64e8c42c86752c570e/imod/mf6/out/disv.py#L63
> 
> The current logic will work alright when xorigin and yorigin are 0 (as imod-python will do), but it fail when somebody has written their model e.g. with flopy with a nonzero origin.
> 
> Note that we're skipping the centroid coordinates as well:
> 
> https://github.com/Deltares/imod-python/blob/afcf5412b7941ce62dc02d64e8c42c86752c570e/imod/mf6/out/disv.py#L64
> 
> 
> https://github.com/Deltares/imod-python/blob/afcf5412b7941ce62dc02d64e8c42c86752c570e/imod/mf6/out/disv.py#L65
> 
> This should be the least controversial since a centroid is a centroid (for a convex polygon), but as far as I remember, xugrid won't respect these coordinates anyway: [Deltares/xugrid#56](https://github.com/Deltares/xugrid/issues/56)

Good point, I've added this and added a test which overrides a DISV file of a simulation, runs it, reads the CBC and checks if the offset in the origin is accounted for.

mjr-deltares commented:
I missed the review request for this issue, sorry 'bout that. (But all good as far as I can tell)
